petab/core.py

```diff
@@ -770,3 +770,66 @@ def measurements_have_replicates(measurement_df: pd.DataFrame) -> bool:
             measurement_df,
             ['observableId', 'simulationConditionId',
              'preequilibrationConditionId', 'time'])).size().values - 1)
+
+
+def flatten_timepoint_specific_output_overrides(
+        petab_problem: Problem) -> None:
+    """If the PEtab problem definition has timepoint-specific
+    observableParameters or noiseParameters for the same observable, replace
+    those by replicating the respective observable.
+
+    This is a helper function for some tools which may not support such
+    timepoint-specific mappings.
+
+    Arguments:
+        petab_problem:
+            PEtab problem to work on
+    """
+
+    df = petab_problem.measurement_df[["observableId", "preequilibrationConditionId", "simulationConditionId"]]
+    df_unique_values = df.drop_duplicates()
+    df_new = pd.DataFrame()
+    for nrow in range(len(df_unique_values.index)):
+        df = petab_problem.measurement_df.loc[(petab_problem.measurement_df['observableId'] >= df_unique_values.iloc[nrow, :].values[0])
+                                         & (petab_problem.measurement_df['preequilibrationConditionId'] <= df_unique_values.iloc[nrow, :].values[1])
+                                         & (petab_problem.measurement_df['simulationConditionId'] <= df_unique_values.iloc[nrow, :].values[2])]
+        df_observable_parameters = df[["observableParameters", "noiseParameters"]]
+        unique_scaling = df_observable_parameters["observableParameters"].unique()
+        unique_noise = df_observable_parameters["noiseParameters"].unique()
+
+        for n_noise in range(len(unique_noise)):
+            for n_scale in range(len(unique_scaling)):
+                idxs = (df["noiseParameters"].str.find(unique_noise[n_noise]) + df["observableParameters"].str.find(
+                            unique_scaling[n_scale]))
+                tmp = df["observableId"].loc[idxs == 0]
```

Comment:
I am afraid that creating new dataframes in a triple-nested loop will scale very badly. We can give it a try though.

Reply:
I agree, that it may not be optimal. Can we still use this as temporary solution? Maybe we can implement an additional issue if this becomes a problem (?)